### PR TITLE
Oh my patient prisoner you've waited for this day and finally

### PR DESCRIPTION
**Pull request by topisani**
you are free, you are free, you are freeeeezing.

Yes, its true.

The whole worlds attention has been steered towards us, as they waited patiently (some not so much) for this to be done.

And finally the day has arrived!

This PR fixes the performance issues with 1.9.4.
Thats right.
Expect a release soon.

~~On the slightly less epic note, @Zeno410, i ported your delayed decoration stuff, but it resulted in only very few chunks actually being populated. I commented it out in `CPRT.populate` and `doPopulate` for now, but if you could take a look at it that would be awesome!~~

**EDIT:** Fixed the delayed decoration stuff, turned out to just be me not porting the eventManager part.
There still seems to be an issue with very rarely some undecorated chunks, but maybe if @whichonespink44 fixes this crash, it will help:

```
java.lang.RuntimeException: DecoHelperRandomSplit is confused.
    at teamrtg.rtg.api.tools.deco.helper.DecoHelperRandomSplit.generate(DecoHelperRandomSplit.java:31)
    at teamrtg.rtg.api.world.gen.RealisticBiomeGenerator.rDecorate(RealisticBiomeGenerator.java:310)
    at teamrtg.rtg.core.world.ChunkProviderRTG.doPopulate(ChunkProviderRTG.java:671)
    at teamrtg.rtg.core.world.ChunkProviderRTG.clearDecorations(ChunkProviderRTG.java:443)
    at teamrtg.rtg.core.world.ChunkProviderRTG.populate(ChunkProviderRTG.java:530)
```

### CODE REVIEW

**Comment by topisani** [comment]
Btw, to get a bit more excited about this, i present to you, a timetable:

```
[04:29:51] [Server thread/INFO]: Preparing start region for level 0
[04:29:52] [Server thread/INFO]: Preparing spawn area: 3%
[04:29:53] [Server thread/INFO]: Preparing spawn area: 8%
[04:29:54] [Server thread/INFO]: Preparing spawn area: 9%
[04:29:55] [Server thread/INFO]: Preparing spawn area: 10%
[04:29:56] [Server thread/INFO]: Preparing spawn area: 12%
[04:29:57] [Server thread/INFO]: Preparing spawn area: 13%
[04:29:58] [Server thread/INFO]: Preparing spawn area: 15%
[04:29:59] [Server thread/INFO]: Preparing spawn area: 17%
[04:30:00] [Server thread/INFO]: Preparing spawn area: 19%
[04:30:01] [Server thread/INFO]: Preparing spawn area: 21%
[04:30:03] [Server thread/INFO]: Preparing spawn area: 23%
[04:30:04] [Server thread/INFO]: Preparing spawn area: 26%
[04:30:05] [Server thread/INFO]: Preparing spawn area: 29%
[04:30:06] [Server thread/INFO]: Preparing spawn area: 32%
[04:30:07] [Server thread/INFO]: Preparing spawn area: 36%
[04:30:08] [Server thread/INFO]: Preparing spawn area: 42%
[04:30:09] [Server thread/INFO]: Preparing spawn area: 47%
[04:30:10] [Server thread/INFO]: Preparing spawn area: 55%
[04:30:11] [Server thread/INFO]: Preparing spawn area: 63%
[04:30:12] [Server thread/INFO]: Preparing spawn area: 71%
[04:30:13] [Server thread/INFO]: Preparing spawn area: 83%
[04:30:14] [Server thread/INFO]: Preparing spawn area: 95%
[04:30:15] [Server thread/INFO]: Changing view distance to 6, from 10
```

The generation itself took 20 secs.

File: mod_deps.properties [hunk 1]
```diff
@@ -1,7 +1,7 @@
 # mod_deps should be 'after:<mod identifier>' (unquoted, semicolon-separated, \escaped for newline, see example below)
-bopver=1.9.4-4.1.0.2037
+bopver=1.9.4-4.1.0.2048
```
Comment (srs-bsns) [inline mod_deps.properties:2]: What was the need to change this? Nothing added to BoP after build 2037 should affect us, so there isn't much need to lock us to the absolute latest BoP build.

File: src/main/java/teamrtg/rtg/core/event/EventManagerRTG.java [hunk 2]
```diff
@@ -33,8 +32,15 @@
 import teamrtg.rtg.core.world.gen.structure.MapGenVillageRTG;
 import teamrtg.rtg.core.world.gen.structure.StructureOceanMonumentRTG;
 
+import java.util.ArrayList;
+import java.util.Random;
+import java.util.WeakHashMap;
```
Comment (srs-bsns) [inline src/main/java/teamrtg/rtg/core/event/EventManagerRTG.java:37]: @Team-RTG/rtg-writers
We all need to get on the same page with imports so they'll stop getting shuffled around everytime someone runs an import optimiser when they alter code.. it just makes the diffs more complex than they need to be.

That being said, fallowing the Team-RTG style rules, java imports should remain at the top.

```
javax.*
java.*
<blank line>
all other imports
<blank line>
static imports
```

**Comment by srs-bsns** [comment]
The changes you made to EventManagerRTG.java in 33c0ca2 is going to conflict with #836. Perhaps you should fetch 1.9.4/event-manager and patch from there in anticipation of #836 being merged.

**Comment by whichonespink44** [comment]
We can go the other way around... I'll merge this one first and then resolve the conflicts on #836 

As for the imports... yes, let's get those sorted as it does make the diffs more difficult to read. I agree with the ordering, but the only thing I'd like to enforce is the use of explicit imports rather than wildcards. On the whole, it doesn't really matter one way or another, but I find myself using the imported classes to jump to the class when Eclipse doesn't let me click on a class name due to wrong method signatures.

**Comment by srs-bsns** [comment]
> but the only thing I'd like to enforce is the use of explicit imports rather than wildcards

Sure. I only use the feature to keep the import blocks leaner, but I generally hide the whole block anyways.
If it helps you with code navigation, then we should banish wildcards for imports. That's an easy fix in IDEA.

File: mod_deps.properties (hunk 1 above)
Comment (topisani) [inline mod_deps.properties:2]: I simply did that to see if the bop compact would be fixed by it. It wasnt

**Pull request merged**